add markdown templates for the Config Doc generator

Athou opened the pull request:
- I could not test `allConfig.qute.md` but it seems that the only difference with `configReference.qute` is an additional for loop
- I could not find a way to add table headers in the middle of the table, I don't think it's possible with Markdown so I just added an additional row with bold text.
- I'm not sure about the section header anchors. I added them anyway because that's what the asciidoc templates do but it feels unusual to have links like that inside the documentation. I'd prefer having headings because Github generates anchors automatically for headings but headings are not supported inside tables. 
 
Another possibility would be to create separate tables for each section. It's a different layout than what the asciidoc templates do but that would solve the headings and anchors issues. 

Something like:

### Section 1
| Configuration property | Type | Default |
|---------------------------|------|---------|
| ... | ... | ... |
| ... | ... | ... |

### Section 2
| Configuration property | Type | Default |
|---------------------------|------|---------|
| ... | ... | ... |
| ... | ... | ... |

What do you think ?

I updated the gist with the output of the PR  https://gist.github.com/Athou/38211a5a6600e97f3daa8fd909dbd99f

Athou commented:
I pushed another commit with sections as headings, and here's a gist with the result for comparison: https://gist.github.com/Athou/01c94d441ba4e00ca784b7e636e8fedf

gsmet commented:
One thing we need to make sure is that the sections are at the end. Otherwise it’s going to be a bit messy. 

Nice you saw this level thing, I wanted to draw your attention on it!

gsmet commented:
OK, so:

- I can confirm sections will be at the end
- you can trigger the all config reference by adding `<generateAllConfig>true</generateAllConfig>` to your plugin config. It shouldn't change anything in your case but at least you can have a look. There's a good chance the extension name will be `Not detected` but that's expected.
- I fixed the issue with the annotations appearing here: https://github.com/quarkusio/quarkus/pull/42930

Sections are fine but it's a bit annoying that all the tables don't have the same width and I don't think you can actually fix that on GitHub unfortunately.

I'll let you decide what you prefer and we can tweak things later depending on the feedback.

Athou commented:
>     I can confirm sections will be at the end

Thanks, I modified the template to process sections last.

>     you can trigger the all config reference by adding `<generateAllConfig>true</generateAllConfig>` to your plugin config. It shouldn't change anything in your case but at least you can have a look. There's a good chance the extension name will be `Not detected` but that's expected.

Thanks, it works, I also modified the `allConfig.qute.md` template.

>     I fixed the issue with the annotations appearing here: [Config Doc - Avoid annotations in primitive type name quarkusio/quarkus#42930](https://github.com/quarkusio/quarkus/pull/42930)

Thanks! As a side note, the link to the Duration javadoc is broken (https://docs.oracle.com/en/java/javase/17/docs/api/java/time/Duration.html). Is it really helpful though with the 🛈 link already pointing to the note at the end of the page? The note has a short explaination about how to use Duration and also points the the parse() method of the javadoc (and that link still works).


> Sections are fine but it's a bit annoying that all the tables don't have the same width and I don't think you can actually fix that on GitHub unfortunately.
> 
> I'll let you decide what you prefer and we can tweak things later depending on the feedback.

I agree it would be better if all tables had the same width, but I still feel it looks better and is more in line with what you'd find on other repos on Github.

This is what it looks like now: https://gist.github.com/Athou/01c94d441ba4e00ca784b7e636e8fedf

gsmet reviewed (commented):
Looks mostly good, I added a few comments/questions.

gsmet commented:
> As a side note, the link to the Duration javadoc is broken (https://docs.oracle.com/en/java/javase/17/docs/api/java/time/Duration.html). Is it really helpful though with the 🛈 link already pointing to the note at the end of the page? The note has a short explaination about how to use Duration and also points the the parse() method of the javadoc (and that link still works).

Ah, they change the base URL to include the module name. I added a commit to fix it here: https://github.com/quarkusio/quarkus/pull/42930 .

We have links for all JDK types. I think it's fine.

Athou commented on devtools/config-doc-maven-plugin/src/main/java/io/quarkus/maven/config/doc/generator/AbstractFormatter.java:
```diff
@@ -31,12 +31,12 @@ public String formatDescription(ConfigProperty configProperty) {
         }
 
-        String description = javadoc(javadocElement.get());
+        String description = javadocElement.get().description();
```
I have a javadoc link in one of my comments:

    See {@link FeedRefreshIntervalCalculator} for details.

Without the change, the javadoc tag was printed verbatim.
With the change, the link is correctly replaced with a markdown code block. Do you have another idea on how to transform it correctly? Now that I think about it, maybe I shouldn't use javadoc links in those comments altogether since they are intended for an audience that won't read source code.

gsmet replied:
Don’t you end up with an Asciidoc link and not a Markdown one?
Where does it point exactly? Because I don’t see how it could find your Javadoc base URL?

I think we should support that and what I was wondering is if we should do it in the Maven plugin. But I’m not sure you have everything to do it.

gsmet replied:
Oh and don't change your Javadoc for now, I'll see if I can improve things. It's easier if I have an example around.

Athou replied:
> Don’t you end up with an Asciidoc link and not a Markdown one?

Neither, I get a code block, both in asciidoc and in markdown:
![image](https://github.com/user-attachments/assets/47d2b529-7391-4839-b3e7-4206115cbddc)

gsmet replied:
Ah OK, I see. You're lucky enough that the generated Asciidoc is also valid Markdown.

Could you reintroduce the `javadoc` methods you removed? I will fix it another way soon. Thanks!

Apart from that I think we're good to go, please ping me when done.

Athou replied:
Sure!

Athou commented on devtools/config-doc-maven-plugin/src/main/resources/templates/asciidoc/default/tags/configProperty.qute.adoc:
```diff
@@ -8,6 +8,7 @@ a|{#if configProperty.phase.fixedAtBuildTime}icon:lock[title=Fixed at build time
 --
 {configProperty.formatDescription.escapeCellContent.or("")}
 {#envVar configProperty /}
+
```
It's a mistake indeed, I'll revert the change, thanks!

Athou commented:
@gsmet All done. Do you want me to squash my commits?

I see in [CONTRIBUTING.md](https://github.com/quarkusio/quarkus/blob/main/CONTRIBUTING.md#coding-guidelines) that it's what's expected.

gsmet commented:
Don't worry, I'll reorganize the commits in my PR and will make you coauthor. I'll ping you when I have something for the Javadoc for you to test.

Note that this is 3.16 material, I won't backport it as it's definitely a new feature.

The rest of the fixes I made thanks to our discussion will be backported.
